tools/scripts/broker_test_consumer.sh

```diff
@@ -0,0 +1,133 @@
+# Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+
+# WSO2 Inc. licenses this file to you under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+# You may obtain a copy of the License at
+
+#    http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+#!/usr/bin/env bash
+
+# variable to store the location of properties file
+properties_file_location="resources/broker_test_consumer.properties"
+
+# variable to store jms destination topic/queue.Default value is queue
+jmx_file_location="test_plan/broker_test_queue_consumer.jmx"
+destination=""
+connection_factory_name=""
+jndi_file_location=""
+
+# get inputs from the user -p <location of the properties file> -d topic/queue
+while getopts "hp:d:t:h:v" OPTION
+do
+     case $OPTION in
+         p)
+            properties_file_location=$OPTARG
+            break
+            ;;
+         d)
+            case $OPTARG in
+                queue)
+                    destination="QueueName"
+                    connection_factory_name="QueueConnectionFactory"
+                    jndi_file_location="resources/jndi_queue.properties"
+                    jmx_file_location="test_plan/broker_test_queue_consumer.jmx"
+                ;;
+                topic)
+                    destination="TopicName"
+                    connection_factory_name="TopicConnectionFactory"
+                    jndi_file_location="resources/jndi_topic.properties"
+                    jmx_file_location="test_plan/broker_test_topic_consumer.jmx"
+                    ;;
+                ?)
+                echo $OPTARG
+                    echo "$OPTARG is an invalid destination.JMS destination should be a queue or a topic"
+                    exit
+                ;;
+            esac
+            break
+            ;;
+         h)
+            help_text="Welcome to ballerina message broker micro-benchmark tool\n\nUsage:\n\t./broker_test_consumer.sh [command].\n\nCommands\n\t-h  ask for help\n\t-p  set the location of the properties file\n\t-d  set jms destination type queue/topic\n"
+            printf "$help_text"
+            exit
+            ;;
+         ?)
+            printf "Invalid command.Run ./broker_test_consumer.sh -h for usage.\n"
+            exit
+            ;;
+     esac
+done
+
+# check for target folder
+if [ ! -e target/ ];
+then
+    mkdir -p target/subscriber
+fi
+
+# hash-map to store user desired parameters
+declare -A user_inputs=(["jmeter_home"]="" ["thread_count"]="1" ["number_of_messages"]="1000000")
```

Comment:
These are the defaults if the paramters are not decalred in the properties file, right? Shall we mention these in the README?

Reply:
Already documented.